### PR TITLE
Rework rust caching in CI

### PR DESCRIPTION
**Pull request by FirelightFlagboy**
the same cache for rust & python jobs
- Build rust code using the same profile for rust & python jobs
- Add job to pre generate rust cache
- Update dependencies on action `actions/cache` to `v3.0.11`

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -129,11 +131,101 @@ jobs:
...
+          set -ex
+          cargo build \
+            --locked \
+            --workspace \
+            --profile ${{ env.CARGO_PROFILE }} \
+            --all-targets
+          cargo build \
+            --locked \
+            --workspace \
+            --profile ${{ env.CARGO_PROFILE }}
```
Comment (vxgmichel) [inline .github/workflows/ci.yml:220]: Why is it called twice?

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -129,11 +131,101 @@ jobs:
...
+        timeout-minutes: 40
```
Comment (vxgmichel) [inline .github/workflows/ci.yml:221]: 40? I would expect the build to be much quicker than that

**Review by vxgmichel — APPROVED** [review]
A couple of questions, LGTM otherwise :+1:

File: .github/workflows/ci.yml (hunk 2 above)
Comment (FirelightFlagboy) [reply .github/workflows/ci.yml:221]: When I was testing, I see the macos runner that could take up to ~30 min to build `libparsec`

File: .github/workflows/ci.yml (hunk 1 above)
Comment (FirelightFlagboy) [reply .github/workflows/ci.yml:220]: It's almost called twice :wink: see the first one as `--all-targets` but not the second one.

When I was testing, even tho we specify `--all-targets` (meaning to build everything (tests and libs)), after that step if I do `cargo build` It was still building/linking something.

Comment (FirelightFlagboy) [reply .github/workflows/ci.yml:220]: see https://github.com/Scille/parsec-cloud/actions/runs/3420302223/jobs/5694945778

Comment (vxgmichel) [reply .github/workflows/ci.yml:220]: that's weird tho, isn't it?

File: .github/workflows/ci.yml (hunk 2 above)
Comment (vxgmichel) [reply .github/workflows/ci.yml:221]: oO

File: .github/workflows/ci.yml (hunk 1 above)
Comment (FirelightFlagboy) [reply .github/workflows/ci.yml:220]: Yes It's weird

Comment (FirelightFlagboy) [reply .github/workflows/ci.yml:231]: Anyway I've added a comment to clarify this step on why we're doing 2 `cargo build`

**Pull request merged**